server/packets.py

```diff
@@ -149,6 +150,40 @@ class Packet:
     packet_data: Any
 
 
+class OsuMessage(TypedDict):
+    sender_name: str
+    message_content: str
+    recipient_name: str
+    sender_id: int
+
+
+class OsuChannel(TypedDict):
+    channel_name: str
+    channel_topic: str
+    channel_user_count: int
+
+
+class OsuMatch(TypedDict):
+    match_id: int
+    match_in_progress: bool
+    mods: int
+    match_name: str
+    match_password: str
+    beatmap_name: str
+    beatmap_id: int
+    beatmap_md5: str
+    slot_statuses: list[int]
+    slot_teams: list[int]
+    per_slot_account_ids: list[int]
+    host_account_id: int
+    game_mode: int
+    win_condition: int
+    team_type: int
+    freemods_enabled: bool
+    per_slot_mods: list[int]
+    random_seed: int
```

Comment:
the packet reader now has methods to read these 3 types, alongside the primitive types like `packet_reader.read_i32()`. these will be useful in our packet handlers to reduce code duplication